zebra/zebra_vxlan.c

```diff
@@ -1413,25 +1413,39 @@ static void zl3vni_remote_rmac_del(struct zebra_l3vni *zl3vni,
 				   struct zebra_mac *zrmac,
 				   struct ipaddr *vtep_ip)
 {
-	struct ipaddr ipv4_vtep;
+	struct ipaddr tmp_ip;
+	struct ipaddr *ipv4_vtep, *ipv6_nh;
+
+	memset(&tmp_ip, 0, sizeof(tmp_ip));
+	if (vtep_ip->ipa_type == IPADDR_V6) {
```

Comment:
Shouldn't we check if IPv4 is _REALLY_ mapped? Like using `IS_MAPPED_IPV6()`.

Reply:
Theoretically, yes. But currently only IPv4 VTEP is supported, and the nexthop is either IPv4 or IPv4-mapped. I think we can keep this assumption for now (like in zebra_vxlan_evpn_vrf_route_add), and handle other cases when adding support for IPv6 VTEP.

Reply:
Yes, but at some level it's a bit safer (when VTEPs could be IPv6?) and more understandable looking at the code.

Reply:
Regarding IPv6 VTEPs, things are different because the addresses can't be simply taken as nexthops of IPv4 routes. We should either allocate intermediate v4 addresses or use dataplane-dependent mechanisms like RTA_VIA (see my comment above).
Anyway, I'm adding this validation.